### PR TITLE
Remove border radius from icon-toggle-pictures

### PR DESCRIPTION
**Pull request by marcoambrosini**
Screenshot from 2020-04-21 10-54-30

### CODE REVIEW

File: core/img/actions/promoted-view.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<rect x="1" y="1" width="13" height="9" rx="1" fill="black"/>
```
Comment (skjnldsv) [inline core/img/actions/promoted-view.svg:2]: fill="black" not needed ;)

**Comment by skjnldsv** [comment]
So, you used a radius of 1, but be aware that the grid one have a radius of .5 :)

**Comment by ma12-co** [comment]
> be aware that the grid one have a radius of .5 :)

Can I remove all this nearly invisible border radiuses and make squares here? It would help a lot also with how the 3 little squares look..

**Comment by skjnldsv** [comment]
> Can I remove all this nearly invisible border radiuses and use straight squares here? It would help a lot also with how the 3 little squares look..

see with Jan, I'm fine with whatever :p

**Comment by jancborchardt** [comment]
> > be aware that the grid one have a radius of .5 :)
> 
> Can I remove all this nearly invisible border radiuses and make squares here? It would help a lot also with how the 3 little squares look..

If it looks good, go ahead. :)

**Review by jancborchardt — APPROVED** [review]
Looks super nice! Seeing how we now have 3 icons without text next to each other, shouldn’t we get the 3-dot menu?
> Start call button | 3-dot menu | Sidebar icon

"Fullscreen" and "Grid view" (shorter wording btw) can be in the 3-dot menu. If we want to, we can even put the sidebar icon in the 3-dot menu as "Participants & settings" or just "Conversation settings".

**Comment by ma12-co** [comment]
> Looks super nice! Seeing how we now have 3 icons without text next to each other, shouldn’t we get the 3-dot menu?

I wouldn't do this Jan because we are going to bring the whole moderation menu as a 3 dot menu in the app content. Plus those 2 controls are very important for the call imo: provided there's enough width, I think we should show them explicitly

**Comment by ma12-co** [comment]
@skjnldsv do you know why the build is failing?

**Comment by skjnldsv** [comment]
Unrelated

**Comment by nickvergessen** [comment]
I would vote that these are very specific icons and they should just go to the Talk app for now.

Also they should be minified hiding the script in build/

**Comment by ma12-co** [comment]
> these are very specific icons 

Only the `promoted-view` icon is specific to talk, the grid is used in other apps too.

**Comment by rullzer** [comment]
> > these are very specific icons
> 
> Only the `promoted-view` icon is specific to talk, the grid is used in other apps too.

fair enough. Then move the promoted view to talk ;)

**Comment by ma12-co** [comment]
Wouldn't be better, for the sake of standardization, to leave it here? I mean, that icon is in the end just 1 big rectangle on top 3 smaller rectangles, it could also be easily a gallery icon, or something else we haven't thought of! What if a  contributor developing an app needs something like this? Isn't it better to provide it in the server instead of having him create his own version?

**Comment by rullzer** [comment]
> Wouldn't be better, for the sake of standardization, to leave it here? I mean, that icon is in the end just 1 big rectangle on top of 3 smaller rectangles, it could also be easily a gallery icon, or something else we haven't thought of! What if a contributor developing an app needs something like this? Isn't it better to provide it in the server instead of having him create his own version?
> 
> Edit: we could also add it to talk for now to simplify backports, but I would leave a version of it here as well.

Just dumping all icons in server is not a solution IMO.
Apps should provide their own icons as much as possible. If not only for the backports. And making sure the icon in server isn't suddenly changed.

If it turns out an icon is used consistently across apps for function X. Then we can always move it to server. Or as part of the vue components or whatever.

But lets do that once it is needed. If some app comes along now and thinks they needit. They can just copy the icon from talk.

Let try to keep the server as clean as we can. And the apps as self contained as we can.

**Comment by rullzer** [comment]
Random brain fart. I don't know how easy/feasible it is. But maybe some npm packages nextcloud/icons would make sense?

That way we still have all the standardized icons. But apps just pick the ones they need?

**Comment by skjnldsv** [comment]
> Random brain fart. I don't know how easy/feasible it is. But maybe some npm packages nextcloud/icons would make sense?

this is planned™ ^^

**Comment by ma12-co** [comment]
Allrighty!

**Comment by ma12-co** [comment]
Done, the only change is now the modification to `icon-toggle-pictures`

File: core/img/actions/toggle-pictures-white.svg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<rect x="1" y="1" width="5.83333" height="5.83333" fill="white"/>
+<rect x="1" y="9.16666" width="5.83333" height="5.83333" fill="white"/>
+<rect x="9.16666" y="1" width="5.83333" height="5.83333" fill="white"/>
+<rect x="9.16666" y="9.16666" width="5.83333" height="5.83333" fill="white"/>
```
Comment (skjnldsv) [inline core/img/actions/toggle-pictures-white.svg:5]: Not ok with this change, you switched from round values to things like .1666, this is not scalled on the pixel :)
```svg
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16"><path d="M1 1h6v6H1zM1 9h6v6H1zM9 1h6v6H9zM9 9h6v6H9z"/></svg>
 ```

Also, we shouldn't need a white icon, we have the scss function for it, this is the reason we deleted all the other ones :wink:

**Review by skjnldsv — CHANGES_REQUESTED** [review]
See comment, please also update the main post screenshots so people know what it is about now ;)

**Comment by jancborchardt** [comment]
> But lets do that once it is needed. If some app comes along now and thinks they needit. They can just copy the icon from talk.

Agreed on that. No premature standardization. :)

File: core/img/actions/toggle-pictures.svg [hunk 3]
```diff
@@ -1 +1,6 @@
-<svg xmlns="http://www.w3.org/2000/svg" version="1.1" viewbox="0 0 16 16" width="16" height="16"><rect rx=".5" ry=".5" height="6" width="6" y="1" x="1"/><rect rx=".5" ry=".5" height="6" width="6" y="1" x="9"/><rect rx=".5" ry=".5" height="6" width="6" y="9" x="9"/><rect rx=".5" ry=".5" height="6" width="6" y="9" x="1"/></svg>
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<rect x="1" y="1" width="6" height="6" fill="black"/>
+<rect x="1" y="9" width="6" height="6" fill="black"/>
+<rect x="9" y="1" width="6" height="6" fill="black"/>
+<rect x="9" y="9" width="6" height="6" fill="black"/>
+</svg>
```
Comment (skjnldsv) [inline core/img/actions/toggle-pictures.svg:6]:
```suggestion
<svg width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
<rect x="1" y="1" width="6" height="6"/>
<rect x="1" y="9" width="6" height="6"/>
<rect x="9" y="1" width="6" height="6"/>
<rect x="9" y="9" width="6" height="6"/>
</svg>
```

File: core/img/actions/toggle-pictures.svg [hunk 4]
```diff
@@ -1 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" version="1.1" viewbox="0 0 16 16" width="16" height="16"><rect rx=".5" ry=".5" height="6" width="6" y="1" x="1"/><rect rx=".5" ry=".5" height="6" width="6" y="1" x="9"/><rect rx=".5" ry=".5" height="6" width="6" y="9" x="9"/><rect rx=".5" ry=".5" height="6" width="6" y="9" x="1"/></svg>
+<svg width="16" height="16" fill="none" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><g fill="#000"><rect x="1" y="1" width="6" height="6"/><rect x="1" y="9" width="6" height="6"/><rect x="9" y="1" width="6" height="6"/><rect x="9" y="9" width="6" height="6"/></g></svg>
```
Comment (skjnldsv) [inline core/img/actions/toggle-pictures.svg:1]: You don't need fill xD
```suggestion
<svg width="16" height="16" fill="none" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><rect x="1" y="1" width="6" height="6"/><rect x="1" y="9" width="6" height="6"/><rect x="9" y="1" width="6" height="6"/><rect x="9" y="9" width="6" height="6"/></svg>
```

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Wait a sec – the border-radius removal we actually don’t want. The rounded style is something we go for across the design, and the corners should stay softened as they are now. Feel free to increase the radius to 1px for example.

**Comment by ma12-co** [comment]
@jancborchardt  please take a look at these icons at 100% . There's no roundness to be seen neither with .5 nor with 1px border radius. We're just making these icons blurry.. It's true that we go for the rounded style but there's no need to do this with icons too imo, and many of the icons we have are have straight edges: 

Screenshot from 2020-04-21 14-38-03

**Comment by jancborchardt** [comment]
@ma12-co I agree that currently it’s inconsistent and this is something we need to work on. But look at the latest icons: details/info, confirm, comment, fullscreen, menu-sidebar, video-off, audio-off.

All of these are rounded or have a majority of rounded elements in them, which is what we are going for, and what is absolutely a difference to them being squared, even on icon size. The others merely need to be adjusted, and we shouldn’t take steps back.

**Comment by ma12-co** [comment]
[:cough:](https://material.io/resources/icons/?style=baseline)

**Comment by jancborchardt** [comment]
> [:cough:](https://material.io/resources/icons/?style=baseline)

We are all for a unified style – for that reason it seems off to remove the rounding which is intentional (and visible) from only 1 icon, no?

Yes, Material Design has an icon set where they have sharp and rounded corners mixed. Let’s look at others as well:
- [SF Symbols from Apple](https://developer.apple.com/design/human-interface-guidelines/sf-symbols/overview/): Uses a lot of rounding of the corners to make their style more approachable, friendly and easily perceivable. (They also round their hardware and app icons as we know.)
- [Bootstrap icons](https://icons.getbootstrap.com/): Even more rounding there
- [Fontawesome](https://fontawesome.com/icons?d=gallery): I don’t particularly like their specific style, but they also round stuff off.

**Comment by skjnldsv** [comment]
> * [Fontawesome](https://fontawesome.com/icons?d=gallery): I don’t particularly like their specific style, but they also round stuff off.

should be avoided, they changed license a while ago iirc, [Forkawesome](https://github.com/ForkAwesome/Fork-Awesome) is the new OSS version

**Comment by ma12-co** [comment]
@jancborchardt  the cough was more of and "Let's use these instead" :) 
They have rounded versions too!
https://material.io/resources/icons/?style=round

**Comment by ma12-co** [comment]
Discussion continues here: https://github.com/nextcloud/nextcloud-vue/issues/1051
